runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -232,6 +232,29 @@ public static long getLogTail(Layout layout, CorfuRuntime runtime) {
         return globalLogTail;
     }
 
+
+    public static long getMaxLogFileSize(Layout layout, CorfuRuntime runtime) {
+        long size = 0;
+
+        Layout.LayoutSegment segment = layout.getLatestSegment();
```

Comment:
I don't think this is correct. When nodes join/leave the latest (i.e. just split) segment usually doesn't reflect the log size because of state transfer. You need to query the head node of every segment, because those nodes contain the "super set"  of log entries.

Reply:
It basically calls the active log servers and get the log size for each server and get the max of them.

Reply:
When u add a new node, the the "latest segment" will only contain that new node, so when u query the log size from the new node, it can be inaccurate because when the node appears in the last segment it doesn't mean that it also replicated all the previous segments.